docs/developers/deprecating.md

```diff
@@ -0,0 +1,45 @@
+#
+
+## What is deprecation?
+
+Whenever a feature of iRODS needs to be removed for whatever reason, we follow a process which first deprecates the feature and then removes it. Here are some possible reasons for deprecating a feature...
+
+ - It has some issue which can only be fixed by replacement.
+ - It has been or is being replaced by a better alternative.
+ - Its purpose is unknown or no longer supported.
+
+In general, users and administrators should expect the following process when a server feature is being deprecated:
+
+ 1. A feature can be declared as deprecated in any minor version.
+ 2. The feature will continue to be supported for all major versions in which it is deprecated (and previous major versions in which the feature exists).
+ 3. The feature can be removed in the next major version, but will not necessarily be removed. It may remain in a deprecated state in the next major version.
```

Comment:
The first statement here is not clicking in my head. _Although the last statement helps a great deal._

I think what you're saying is ... A change in major versions does not mean a deprecated feature will be removed. Complete removal is optional across major versions.

Am I interpreting the first statement of (3) correctly?

Reply:
How's this...

"The feature can be removed in the next major version after deprecation, but the Consortium reserves the right to not remove it, leaving it in a deprecated state in the next major version."

Reply:
I like it. It's much clearer to me.

Reply:
Done